pages/api/zapbye.ts

```diff
@@ -0,0 +1,100 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import jsforce from 'jsforce';
+
+const CONSTANTS = {
+      SALESFORCE_EMAIL: process.env.SALESFORCE_EMAIL,
+      SALESFORCE_PASSWORD: process.env.SALESFORCE_PASSWORD,
+      SALESFORCE_SECURITY_TOKEN: process.env.SALESFORCE_SECURITY_TOKEN,
+      LEMLIST_API_KEY: process.env.LEMLIST_API_KEY,
+      SALESFORCE_TASK_SUBJECT: 'Lemlist Email',
+      SALESFORCE_LEMLISTTASK_ID: '0123k0000014TCyAAM',
+      SALESFORCE_STATUS: 'Completed',
+      SALESFORCE_TASK_SUBTYPE: 'Task',
+      SALESFORCE_SOBJECT_TYPE: 'Task',
+      LEMLIST_CAMPAIGN_FETCH_COOLDOWN_TIME_IN_MS: 30000,
+};
+
+let salesforceConnection;
+let cachedCampaignsArray;
+let lastCampaignCheckedTimestamp;
+
+export default async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+      const {
+            method,
+      } = req;
+
+      if (method != "POST") {
+            res.statusCode = 405;
+            res.json({ error: "Method Not Allowed" });
+            return;
+      }
+
+      if(!salesforceConnection || !salesforceConnection.accessToken) {
+            console.log("Attempting to log in!");
+            salesforceConnection = new jsforce.Connection();
+            await salesforceConnection.login(CONSTANTS.SALESFORCE_EMAIL, CONSTANTS.SALESFORCE_PASSWORD + CONSTANTS.SALESFORCE_SECURITY_TOKEN, function(err, res) {
+                  if (err) {
+                        salesforceConnection = null;
+                        res.statusCode = 500;
+                        res.body = err;
+                        return console.error(err);
+                  }
+            });
+      }
+      let lemlistPayload = req.body;
+      await salesforceConnection.sobject(CONSTANTS.SALESFORCE_SOBJECT_TYPE).create({
+            TaskSubtype: CONSTANTS.SALESFORCE_TASK_SUBTYPE,
+            Subject: CONSTANTS.SALESFORCE_TASK_SUBJECT,
+            Status: CONSTANTS.SALESFORCE_STATUS,
+            RecordTypeId: CONSTANTS.SALESFORCE_LEMLISTTASK_ID,
+            ActivityDate: new Date(), //DueDate
+            Lemlist_Campaign_Name__c: await getLemlistCampaignNameFromCampaignId(lemlistPayload.campaignId),
+            Lemlist_Sender__c: lemlistPayload.leadFirstName + ' ' + lemlistPayload.leadLastName,
+            Lemlist_Email__c: lemlistPayload.leadEmail,
+            Lemlist_Task_Type__c: lemlistPayload.type,
+            Lemlist_Sequence_Step_Number__c: lemlistPayload.sequenceStep,
+      },{},function(err, ret) {
```

Comment:
```suggestion
      }, {}, (err, ret) => {
```

Reply:
the spacing (not the function change) is usually picked up by our linter prettier:

You can install here: https://prettier.io/docs/en/webstorm.html
Here are our default params for it

`--write $FilePathRelativeToProjectRoot$ --single-quote --trailing-comma es5 --print-width 120`

Reply:
Same thing here, default to handling the promise rather then using the callback.

Reply:
Got it. Configured my standard prettier to run on save and reformat, added the external prettier, and also configured prettier for the project via package.json.